code/modules/mob/living/living_defense.dm

```diff
@@ -165,6 +165,10 @@
 	. = ..()
 	set_light(0) //Reset lighting
 
+/mob/living/carbon/xenomorph/caste/boiler/ExtinguishMob()
```

Comment:
Might be better to update the original xeno proc to just check `isxenoboiler` to avoid destroying and recreating a light source.

Either way this fixes the issue.

Reply:
I did that first but Roh didn't like it

Reply:
PsyKzz brings a good point from an optimization perspective, but from a maintainability one I believe this is better.
A third option to take the best of both worlds would be either a variable that tracks the natural lighting, or even better a light-emitting component.
Those ideas are way out of scope and this fixes the issue. If you want to learn how to do them, poke me on discord. Else this is very approved.

Reply:
This type path is incorrect.

Reply:
goddamnit